### ISSUE TITLE
Standardize link text and URL terminology across the editor

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59993]
### Description

Links and navigation links / submenus can be edited in the editor via a few components.

Although these user interfaces come from different components, the terminology should be consistend when it comes to:
- The link text.
- The link URL

RIght now, the names of these fields are different and don't contribute to UI clarity, especially when they can be displayed at the same time under the eyes of the users.

- The link editing UI uses the terms 'text' and 'link'.
- The block inspector for navigation links and submenus uses the terms 'label' and 'url'.

I'd vote for what makes most sense for users and I'd think the most familiar terminology for users is text / link.
Regardless, the UI should use the same terminology everywhere.

Screenshot when editing a page link in a navigation:

![Screenshot 2024-03-19 at 14 24 52](https://github.com/WordPress/gutenberg/assets/1682452/f5af32ce-fa32-4053-b26c-82b46a313013)

Screenshot when editing a custom link in a navigation:

![Screenshot 2024-03-19 at 14 25 56](https://github.com/WordPress/gutenberg/assets/1682452/068171b9-6bf6-46a7-96f5-d4778eac6acb)

The same applies to navigation submenus.

### Step-by-step reproduction instructions

- Go to the site editor > Design >  Navigation >  any menu
- Make sure the Settings panel is open.
- Edit a link by clicking on it, then click 'Link' in the block toolbar and then the 'Edit link' pencil icon button.
- Observe the link editing UI in the popover uses the terms Text / Link while the block inspector uses Label / URL.
- Repeat for various link types: Page link, Custom link...
- Repeat for a navigation submenu.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Some of the relevant files to be checked:
```
packages/block-editor/src/components/link-control/index.js
packages/block-editor/src/components/link-control/search-input.js
packages/block-library/src/navigation-link/edit.js
packages/block-library/src/navigation-submenu/edit.js
```

**Comment by richtabor** [comment]
I had this queued up to write an issue for as well. Yes, we should tweak this.

> I'd vote for what makes most sense for users and I'd think the most familiar terminology for users is text / link.
Regardless, the UI should use the same terminology everywhere.

Yes, "Text" and "Link" feel appropriate to me. 

We could potentially do the same for the social links as well: https://github.com/WordPress/gutenberg/pull/60047. Perhaps one day it uses LinkControl as well even.

**Comment by t-hamano** [comment]
What do you think about following the core classic editor and using the term URL/Link Text?

![image](https://github.com/WordPress/gutenberg/assets/54422211/2586e9f0-ea90-4bd1-a552-e3c293e55dc6)

**Comment by richtabor** [comment]
> What do you think about following the core classic editor and using the term URL/Link Text?

Sorry, missed this comment before https://github.com/WordPress/gutenberg/pull/60116. 

I think what we have in #60116 is sufficient, and now at least they are consistent across.

**Comment by afercia** [comment]
> I think what we have in https://github.com/WordPress/gutenberg/pull/60116 is sufficient

I don't think it's sufficient.

- in https://github.com/WordPress/gutenberg/pull/60047#discussion_r1535522540 I suggested to just use the terminology WordPress always used: 'Link text'.
- @t-hamano [suggestd](https://github.com/WordPress/gutenberg/issues/59993#issuecomment-2015250410) the same here in a previous comment.
- 'Link text' is the terminology WordPress users are familiar with since ages, I'm not sure there's a good reason to change it.
- 'Text' only isn't sufficient because it doesn't clarify the context. What this 'text' is? While in _some_ cases the UI may provide context, in other cases it doesn't. For example, in the block inspector Page link, Custom links, Submenu, clarify the context in the block card at the top, in a way. However, the Social link card doesn't. See screenshot below.
- When removing the link text in a navigation block, the placeholder text of the contenteditable still says 'Add label...'. This eas missed in #60116 and must to be chanced. See second screenshot.

Screenshots of the block inspector cards with various link-related blocks, to illustrate the term 'Text' doesn't clarify sufficiently what the input field is about:

![Screenshot 2024-03-25 at 08 40 09](https://github.com/WordPress/gutenberg/assets/1682452/88d674c2-fcb1-4024-81c6-c1f6cdb55c43)

Placeholder text of the navigation items still says 'Add llabel...':

![Screenshot 2024-03-25 at 08 52 34](https://github.com/WordPress/gutenberg/assets/1682452/c6b3c765-f1db-4cdd-b483-988927dac35a)

Reopening to allow for broader discussion.

**Comment by t-hamano** [comment]
This is an additional suggestion. For example, the core navigation menu consistently uses the term "URL". On the other hand, the text to which the URL is applied seems to use the terms "Link Text" and "Navigation Label". Could this basic rule also be applied to Gutenberg?

![classic-nav](https://github.com/WordPress/gutenberg/assets/54422211/48152529-eb41-40fd-8258-7a5007a96b23)

**Comment by afercia** [comment]
Yep, that 'Navigation Label' in the classic menus is pretty old. It's inconsistent with custom links and also a little weird to translate. Also, it shouldn't be title case. Not sure changing such an old UI is worth it but, in case, I'd change it to 'Link text'.

Re: the term URL I'm not sure I have strong opinions.
- On one hand, I'd prefer to keep it as 'URL' because that is what WordPress always used and users are familiar with it.
- On the other hand, URL may be a bit too technical for some users. I'd think it's way more common in everyday language to just say 'Link'.

**Comment by t-hamano** [comment]
Regarding the inline link, I would like to provide two more examples outside of WordPress.

Notion: "Page or URL" and "Link title"

![image](https://github.com/WordPress/gutenberg/assets/54422211/154c539e-572d-4aea-8c42-113ced99730b)

Slack: "Text" and "Link" (This is consistent with the changes made in #60116)

![image](https://github.com/WordPress/gutenberg/assets/54422211/f489b80b-1478-415b-8fd4-ef60121aa3cb)

To be honest, I'm also wondering which is the correct answer in WordPress 😅

**Comment by richtabor** [comment]
> in https://github.com/WordPress/gutenberg/pull/60047#discussion_r1535522540 I suggested to just use the terminology WordPress always used: 'Link text'.

Link text is _fine_ in that particular context, but if we do [what I suggested](https://github.com/WordPress/gutenberg/pull/60047#discussion_r1534275275)—to bring the link field there as well—then "Text" and "Link" would work great there as well. 

"Link text" and "Link" adds more words with not enough benefit for the increased word count. 

> On the other hand, URL may be a bit too technical for some users. I'd think it's way more common in everyday language to just say 'Link'.

Agreed. "URL" is a technical term—I'd avoid using terminology that requires additional understanding. 

> When removing the link text in a navigation block, the placeholder text of the contenteditable still says 'Add label...'. This eas missed in https://github.com/WordPress/gutenberg/pull/60116 and must to be chanced. See second screenshot.

This should be "Link" (no ellipsis) — to follow suite of other placeholder text.

**Comment by Niyatijain-9** [comment]
Hey I wanted to work on the same. Kindly assign me this @afercia

**Comment by afercia** [comment]
@Niyatijain-9 thanks for willing to work on this. You don't need to be assigned to an issue to work and submit a PR.

**Comment by yukta8** [comment]
Hello! I came across this issue and found it interesting. Is this still open to work on?
If so, I'd be happy to take it up and submit a PR. Please let me know if I can proceed
Thanks!

**Comment by afercia** [comment]
I think https://github.com/WordPress/gutenberg/pull/60116 took care of this but failed to properly reference this issue to close it.
Noting that in a following PR https://github.com/WordPress/gutenberg/pull/65458 the label of the LinkControl was changed to `Search or type URL` to match the aria-label and because that is what the input field does. The other input in the settings panel is labeled 'LInk' because there's no search functionality there. Screenshot:

![Image](https://github.com/user-attachments/assets/3e3dfd62-00a8-49cf-80d1-16ca8660b614)
